src/test/controller/BaseController.ts

```diff
@@ -99,16 +104,30 @@ export class BaseController {
   };
 
   /**
-   * Create a mock authenticated user
+   * Create a mock authenticated user.
+   * * Note this should be called in the beforeEach of test files (we do not want tests to rely on the same user in multiple tests)
    * */
   async createAuthenticatedUser(): Promise<void> {
+    const authenticatedUser = new UserFactory().generate();
+    await getConnection()
+      .getCustomRepository(UserRepository)
+      .save(authenticatedUser);
     // create user
     const user = await (this.mockFirebaseAdmin as any).createUser({
-      email: 'test@gmail.com',
-      password: 'testpassword',
+      uid: authenticatedUser.id,
+      email: randomString(20) + '@gmail.com',
+      password: randomString(20),
     });
     this.idToken = await user.getIdToken();
-    this.firebaseId = user.uid;
+    this.authenticatedUser = authenticatedUser;
+  }
+
+  /**
+   * Delete the mock authenticated user
+   * * Note this should be called in the afterEach of test files (we do not want tests to rely on the same user in multiple tests)
+   * */
+  async deleteAuthenticatedUser(): Promise<void> {
+    await this.mockFirebaseAdmin.deleteUser(this.authenticatedUser.id);
   }
```

Comment:
I don't like how future devs working on controller tests will need to be responsible for rememberingg to specify these two methods in `beforeEach` and `afterEach`. Will think about this some more...